hamilton/src/hybrid_planner.cpp

```diff
@@ -0,0 +1,547 @@
+#include "hamilton/hybrid_planner.h"
+
+//TODO namespace
+//TODO bools aren't returning anything!
+namespace hamilton
+{
+	
+	HybridPlanner::~HybridPlanner()
+	{
+		moveit_group_ptr_.reset();
+		kinematic_state_.reset();
+		//TODO Descartes
+	}
+
+	void HybridPlanner::init()
+	{
+		// Instantiating a robot model
+		descartes_robot_model_ptr_.reset(new descartes_moveit::MoveitStateAdapter);
+		// TOCHECK as compared to: 
+		//descartes_core::RobotModelPtr model(new descartes_moveit::MoveitStateAdapter);
+		//TOCHECK or as in Godel
+		// init descartes with kinematic_state_ ? 
+		//   descartes_core::RobotModelPtr createRobotModel(const moveit::core::RobotStatePtr robot_state,
+                                                 // const std::string& group_name,
+                                                 // const std::string& tool_frame,
+                                                 // const std::string& world_frame,
+                                                 // uint8_t iterations)
+		if(descartes_robot_model_ptr_->initialize(ROBOT_DESCRIPTION_PARAM,
+		                              config_.group_name,
+		                              config_.world_frame,
+		                              config_.tip_link))
+		{
+		ROS_INFO_STREAM("Descartes Robot Model initialized");
+		}
+		else
+		{
+		ROS_ERROR_STREAM("Failed to initialize Robot Model");
+		exit(-1);
+		}
+
+		// TODO add planner type as config parameter?
+		bool succeeded_dense = dense_planner_.initialize(descartes_robot_model_ptr_);
+		if(succeeded_dense)
+		{
+		ROS_INFO_STREAM("Descartes dense planner initialized");
+		}
+		else
+		{
+		ROS_ERROR_STREAM("Failed to initialize Descartes dense Planner");
+		exit(-1);
+		}  
+
+		bool succeeded_sparse = sparse_planner_.initialize(descartes_robot_model_ptr_);
+		if(succeeded_sparse)
+		{
+		ROS_INFO_STREAM("Descartes sparse planner initialized");
+		}
+		else
+		{
+		ROS_ERROR_STREAM("Failed to initialize Descartes sparse Planner");
+		exit(-1);
+		}
+
+		ROS_INFO_STREAM("Descartes members initialized");
+
+		// creates publisher for trajectory visualization
+		marker_publisher_  = nh_.advertise<visualization_msgs::MarkerArray>(VISUALIZE_TRAJECTORY_TOPIC,1,true);
+
+		//Initializes MoveIt!
+		moveit_group_ptr_ = MoveGroupPtr(new moveit_group_interface::MoveGroup(config_.group_name));
+		//TOCHECK TODO doesn't compile
```

Comment:
Check comment in the member variables of header file. "Too many member variables. kinematic_state_ and joint_model_group_ should have been accessed by descartes::MoveItStateAdapter."
http://docs.ros.org/indigo/api/descartes_moveit/html/classdescartes__moveit_1_1MoveitStateAdapter.html
Sorry for the misnomer, by descartes adapter I was referring to descartes_moveit::MoveitStateAdapter, which has
```
std::string 	group_name_ 
planning_scene::PlanningScenePtr 	
planning_scene_robot_model_loader::RobotModelLoaderPtr 	robot_model_loader_
robot_model::RobotModelConstPtr 	robot_model_ptr_
moveit::core::RobotStatePtr 	robot_state_
std::vector< std::vector< double > > 	seed_states_
```
as member variables. 

Now, in struct HybridConfiguration, I included `std::vector<double> seed_pose`(which should have been `seed_poses`) and in the member variables I have:
```
moveit::core::RobotStatePtr kinematic_state_;   
robot_model::RobotModelConstPtr& kinematic_model_;
const moveit::core::JointModelGroup* joint_model_group_;
```
This seems a bit of overkill. Maybe I should do a :
```
  descartes_core::RobotModelPtr createRobotModel(const moveit::core::RobotStatePtr robot_state,
                                                 const std::string& group_name,
                                                 const std::string& tool_frame,
                                                 const std::string& world_frame,
                                                 uint8_t iterations)
  {
    using descartes_core::RobotModelPtr;

    RobotModelPtr ptr = robot_model_loader.createInstance(PLUGIN_NAME);
    ptr->initialize(robot_description, group_name, world_frame, tool_frame);
    return ptr;
  } 
```  
see https://github.com/ros-industrial-consortium/godel/blob/hydro-devel/godel_path_planning/src/trajectory_planning.cpp#L27-38

and
```
kinematic_state_ = moveit_group_ptr_->getCurrentState();
kinematic_state_->setToDefaultValues(); //TOCHECK
kinematic_model_ = kinematic_state_->getRobotModel();
joint_model_group_ = kinematic_model_->getJointModelGroup(config_.group_name); 
```
should be called in `appendFreeSpaceSegment()`(https://github.com/madratman/hamilton/blob/indigo-devel/hamilton/src/hybrid_planner.cpp#L308-334) instead of `init()`(https://github.com/madratman/hamilton/blob/indigo-devel/hamilton/src/hybrid_planner.cpp#L86-87) as `update()